src/Database/Log/LoggedQuery.php

```diff
@@ -78,8 +78,8 @@ protected function interpolate(): string
             }
 
             if (is_string($p)) {
-                // Likely binary data.
-                if (!ctype_print($p)) {
+                // Likely binary UUID.
+                if (strlen($p) === 16 && !ctype_print($p)) {
```

Comment:
I think the only complete solution is regex. This will limit all other non-ascii characters as well which will convert a LOT of non-basic english strings even if we replaced all the whitespace characters before checking.

> Only ascii 32 thru 126 (inclusive) are considered printable

Reply:
Another approach could be to use a pattern like `'/[\x00-\x1F\x7F]/u'` and see if the results are different than the original value. If they are we can infer the string was bytes.

Reply:
Do we have a test that logs a couple utf-8 unicode characters?

Reply:
> Do we have a test that logs a couple utf-8 unicode characters?

Can you add what you have in mind? Or post such a string here that I can test it?
The current solution now here already looks less problematic than the check before.

Reply:
It's fine. I leave it to you two :)

Reply:
Strings like https://www.soscisurvey.de/tools/view-chars.php work fine.